resourcescaler.go

```diff
@@ -113,82 +235,244 @@ func (s *AppResourceScaler) patchIguazioTenantAppServiceSets(namespace string, j
 	return nil
 }
 
-func (s *AppResourceScaler) waitForServiceReadiness(namespace string, serviceName string) error {
-	s.logger.DebugWith("Waiting for service readiness", "serviceName", serviceName)
+func (s *AppResourceScaler) waitForServiceState(namespace string, serviceName string, state string) error {
+	s.logger.DebugWith("Waiting for service to reach state", "serviceName", serviceName, "state", state)
 	for {
```

Comment:
This can be stuck forever if something goes wrong (like a service is restarted or goes into error state mid scaling to/from zero.
Suggest some reasonable long timeout and using a `select` with a ticker and a deadline